### PR TITLE
feat: new python workflow, new golang test suite and improvements

### PR DESCRIPTION
**Pull request by alexrodfe**
## Why
We wanted to have an additional workflow containing a python node to add more examples to our training repo. We also wanted to be able to send multiple requests from our training examples to aid our testing.

## What
- New python node 'echo'
- New python workflow
- New golang test suite
- Updated chronograph dashboard
- Now proto files will be in dedicated 'proto' folder
- Improvements

### CODE REVIEW

**Comment by aacecandev** [comment]
I'd really love to have functions commented since this is a training repository

**Comment by alexrodfe** [comment]
Good idea @aacecandev ! I have added documentation regarding full disclosure of the whole repo. Hope it is useful.

File: descriptor/src/etl/main.go [hunk 1]
```diff
@@ -2,6 +2,7 @@ package main
 
 import (
 	"fmt"
+	localProto "main/proto"
 
 	"github.com/golang/protobuf/proto"
```
Comment (JYisus) [inline descriptor/src/etl/main.go:7]: @alexrodfe maybe we can use `protobuf` as alias for this package

File: test_go/main.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+// increaseTotalCounter is an auxiliary func that uses a mutex so all gorutines can share
+// a common counter, upon calling the counter will increase.
+func increaseTotalCounter() {
+	counterMutex.Lock()
+	totalRequests++
+	counterMutex.Unlock()
+}
+
+// the goal of this go executable is to emulate several clients making requests at the same time
```
Comment (JYisus) [inline test_go/main.go:33]: @alexrodfe In my opinion, this description should be in the README of the repo.

**Review by JYisus — CHANGES_REQUESTED** [review]

**Comment by JYisus** [comment]
> I'd really love to have functions commented since this is a training repository

@aacecandev @alexrodfe The main functions are documented in the wiki, maybe we should link the wiki in the README, but having the code overcommented is a bad practice. The code must be enough clear to explain itself, and comments make the code hard to maintain (because you have to maintain the code and the comments). I'd keep comments of the helpers functions that have some complexity, but the handler, handlerInit and all the context functions are already explained in the wiki.

File: descriptor/krt.yml [hunk 3]
```diff
@@ -2,7 +2,7 @@ version: descriptor-v1
 description: Version for testing.
 entrypoint:
   proto: public_input.proto
-  image: konstellation/kre-entrypoint:latest
+  image: konstellation/kre-entrypoint:1.8.0
```
[the remark this answers is not in the document]
Comment (alexrodfe) [reply descriptor/krt.yml:5]: Oh, alright seems fine by me :+1:

File: descriptor/src/etl/main.go [hunk 4]
```diff
@@ -2,6 +2,7 @@ package main
 
 import (
 	"fmt"
+	localProto "main/proto"
```
[the remark this answers is not in the document]
Comment (alexrodfe) [reply descriptor/src/etl/main.go:5]: I don't think its relevant to where the proto is generated, we are still using several packages being named/related to proto at the same time.
Maybe having the protobuf/proto package renamed to protobuf and the local proto to proto is clearer.

File: descriptor/src/etl/main.go [hunk 5]
```diff
@@ -10,46 +11,59 @@ import (
...
 func handler(ctx *kre.HandlerContext, data *any.Any) (proto.Message, error) {
 	ctx.Logger.Info("[handler invoked]")
 
-	req := &Request{}
-	res := &EtlOutput{}
+	req := &localProto.Request{}
+	res := &localProto.EtlOutput{}
 
+	// we need to unmarshal recieved data to our expected request struct type
 	err := anypb.UnmarshalTo(data, req, proto2.UnmarshalOptions{})
```
[the remark this answers is not in the document]
Comment (alexrodfe) [reply descriptor/src/etl/main.go:30]: proto2 for this main.go is the "google.golang.org/protobuf/proto" package

File: test_go/main.go (hunk 2 above)
Comment (alexrodfe) [reply test_go/main.go]: Seems like a good idea, I have updated the readme.

**Comment by alexrodfe** [comment]
Agreed with @JYisus resolution. I have deleted comments explained in the docs and maintained comments for those auxiliary funcitons.

File: descriptor/src/etl/main.go (hunk 1 above)
Comment (alexrodfe) [reply descriptor/src/etl/main.go]: With the new refactor of the golang runner I have set new aliases. As we now only have 2 proto packages the one coming from google will be named alias 'protobuf' and our local generated proto will stay as alias 'proto'.

**Comment by alexrodfe** [comment]
I am aware I left replace instructions in the go.mod files. I will delete those once we have merged our new kre-go runners updated with the new proto packages.

**Comment by JYisus** [comment]
> I am aware I left replace instructions in the go.mod files. I will delete those once we have merged our new kre-go runners updated with the new proto packages.

Okay, but you have to delete this lines before merge

**Review by JYisus — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -2,11 +2,27 @@
...
-## Tests
+## How to send manual requests to your deployed versions
 
-Here are some examples:
-./scripts/run_test.sh kre descriptor-v1 GoDescriptor 2
+Here are some examples:  
+
+``` sh
+./scripts/run_test.sh kre descriptor-v1 GoDescriptor 2  
 ./scripts/run_test.sh kre descriptor-v1 GoDescriptor 2 Alex
```
Comment (JYisus) [inline README.md:16]: @alexrodfe  i would put as an example a real repo instead of Alex

**Review by JYisus — CHANGES_REQUESTED** [review]

File: descriptor/krt.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (aacecandev) [reply descriptor/krt.yml]: I totally disagree with this @JYisus. From almost every POV, using latest is a bad practice.

**Review by aacecandev — APPROVED** [review]
Good work guys, I like the PR. Comments are very helpful to those new to Go and KRE like me

**Review by JYisus — APPROVED** [review]

**Pull request merged**